src/session-pool.ts

```diff
@@ -752,41 +758,54 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
     let nReturned = 0;
     const nRequested: number = amount;
 
-    // while we can request as many sessions be created as we want, the backend
-    // will return at most 100 at a time, hence the need for a while loop.
-    while (amount > 0) {
-      let sessions: Session[] | null = null;
+    if (!this.database._observabilityOptions) {
```

Comment:
Why do we need to set. the observability option here again ? We should be doing it in constructor ?

Reply:
There was some test case in which the observability options for database weren't set despite having been passed into the Spanner constructor.

Reply:
Then we should find a way to fix those test cases and not add the code here again.